torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -3097,7 +3100,9 @@ c10::intrusive_ptr<Work> ProcessGroupNCCL::pointToPoint(
   // is gpuGuard needed for the if block below, or can i swap them
   at::cuda::OptionalCUDAGuard gpuGuard;
 
-  if (enableNanCheck_) {
+  // Only check for NaN for send ops, for recv ops `tensor` can be a random
+  // placeholder
+  if (enableNanCheck_ && opType == OpType::SEND) {
```

Comment:
double checking- does this check work correctly for coalesced p2p too? would this function be called when creating the individual isend ops and check their nans at that time?

Reply:
Yeah, I think individual ops in batch P2P would call into this function too, with OpType::Send or OpType::Recv, so they should be covered.